src/uglify/index.js

```diff
@@ -0,0 +1,77 @@
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import findCacheDir from 'find-cache-dir';
+import workerFarm from 'worker-farm';
+import { get, put } from './cache';
+import { encode } from './serialization';
+import versions from './versions';
+
+let workerFile = path.join(__dirname, 'worker.js');
+
+try {
+  const testWorkerFile = path.join(__dirname, '..', '..', 'dist', 'uglify', 'worker.js');
+  fs.accessSync(workerFile);
+  workerFile = testWorkerFile;
+} catch (e) { } // eslint-disable-line no-empty
+
+class Uglify {
+  constructor({
+    cache = findCacheDir({ name: 'uglifyjs-webpack-plugin' }),
+    workers = Math.max(os.cpus().length - 1, 1),
```

Comment:
Hmm, this isn't quite right. If the user has  1 cpu, we want `workers` to be `0`, and use that as a signal to not fork any new processes, as that'll be much slower if we forked when there is only one CPU.